.github/workflows/wheels.yml

```diff
@@ -147,3 +147,63 @@ jobs:
           # https://anaconda.org/multibuild-wheels-staging/numpy
           # The tokens were originally generated at anaconda.org
           upload_wheels
+  build_sdist:
+    name: Build sdist
+    needs: get_commit_message
+    if: >-
+      contains(needs.get_commit_message.outputs.message, '[wheel build]') ||
+      github.event_name == 'schedule' ||
+      github.event_name == 'workflow_dispatch' ||
+      (github.event_name == 'pull_request' &&
+      (contains(github.event.pull_request.labels.*.name, '36 - Build') ||
+      contains(github.event.pull_request.labels.*.name, '03 - Maintenance') ||
```

Comment:
I see this is done for wheels too, and rationale given is:
```
# 03 - Maintenance(for dependency upgrades), and
```
This rationale is not correct; the total number of PRs with this label is ~2250. It can be safely dropped, dependency upgrades that are of interest are not coming in through PRs often (e.g., compilers, or distro BLAS). And if something goes wrong there, it will show up soon enough in the scheduled builds.